src/main/java/com/Incubatormsrv/Incubatormsrv/weather/service/WeatherService.java

```diff
@@ -0,0 +1,62 @@
+package com.Incubatormsrv.Incubatormsrv.weather.service;
+
+import _Users_luyanda_glucode.com_Documents_GitHub_Incubator_msrv.api.WeatherApi;
+import _Users_luyanda_glucode.com_Documents_GitHub_Incubator_msrv.model.WeatherResponse;
+import okhttp3.OkHttpClient;
+import okhttp3.Request;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Component;
+import retrofit2.Response;
+import retrofit2.Retrofit;
+import retrofit2.converter.gson.GsonConverterFactory;
+
+import java.io.IOException;
+
+@Component
+public class WeatherService {
+
+    private final String baseUrl;
+    private final String key;
+    private final String host;
+    private WeatherApi weatherApi;
+    public WeatherService(@Value("${weather.X-RapidAPI-Key}") String key,
+                          @Value("${weather.X-RapidAPI-Host}") String host,
+                          @Value("${weather.baseUrl}") String baseUrl){
+        this.key = key;
+        this.baseUrl = baseUrl;
+        this.host = host;
+        this.weatherApi = retrofit().create(WeatherApi.class);
+    }
+
+    private Retrofit retrofit() {
```

Comment:
Please move this to your configuration and inject the interface to your service wrapper class.